src/main/java/taxi/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java

```diff
@@ -0,0 +1,144 @@
+package taxi.dao.jdbcimpl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import taxi.dao.ManufacturerDao;
+import taxi.exception.DataProcessingException;
+import taxi.lib.Dao;
+import taxi.model.Manufacturer;
+import taxi.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoJdbcImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers "
+                + "(manufacturer_name, manufacturer_country)"
+                + " VALUES (?, ?)";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            String name = manufacturer.getName();
+            String country = manufacturer.getCountry();
+            preparedStatement.setString(1, name);
+            preparedStatement.setString(2, country);
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject(1, Long.class));
+            }
+            preparedStatement.close();
+            resultSet.close();
```

Comment:
if you add prepareStatement to try with resources then this is redundant

Reply:
removed redundant `closes()` at the moment when moved `preparedStatements` inside the `try` block